pages/api/transactions/index.js

```diff
@@ -1,21 +1,34 @@
 import dbConnect from "@/db/connect.js";
 import Transaction from "@/db/models/Transaction";
+import { getServerSession } from "next-auth/next";
+import { authOptions } from "../auth/[...nextauth]";
 
 export default async function handler(request, response) {
-	await dbConnect();
-	try {
-		if (request.method === "GET") {
-			const transactions = await Transaction.find();
-			return response.status(200).json(transactions);
-		}
+  const session = await getServerSession(request, response, authOptions);
+  await dbConnect();
+  try {
+    if (!session) {
+      response.status(401).json({ status: "Not authorized" });
+      return;
+    }
 
-		if (request.method === "POST") {
-			const transactionData = request.body;
-			await Transaction.create(transactionData);
-			return response.status(201).json({ status: "Transaction added." });
-		}
-	} catch (error) {
-		console.error("Error fetching transactions:", error.message);
-		return response.status(400).json({ error: "Internal Server Error" });
-	}
+    if (request.method === "GET") {
+      const transactions = await Transaction.find({
+        author: session.user.email,
+      });
+      return response.status(200).json(transactions);
```

Comment:
please separate the resturn and the response into two lines. Please do that for each endpoint.